openedx-unsupported/configuration
# SE-1841 fix oauth retirement user provision

Currently, there are both `EDXAPP_RETIREMENT_SERVICE_USER_NAME` and `EDXAPP_RETIREMENT_SERVICE_WORKER_USERNAME` used for what appears to be the same thing (please correct me if I'm wrong). 

We believe that this is what is causing [this ansible task](https://github.com/edx/configuration/blob/42431cf549a1e74b2892bd71e9bafc5f412b52dd/playbooks/roles/oauth_client_setup/tasks/main.yml#L71-L90) (create oauth configuration) to fail on the [retirement service entry](https://github.com/edx/configuration/blob/42431cf549a1e74b2892bd71e9bafc5f412b52dd/playbooks/roles/oauth_client_setup/defaults/main.yml#L94-L99) with the default configuration and retirement service enabled (`COMMON_RETIREMENT_SERVICE_SETUP: true`).

Also having two different variables for the same thing is confusing.

This PR renames all remaining instances of the former to the latter.

For background on these recent retirement changes, please see:     https://github.com/edx/configuration/pull/5040 and     https://github.com/edx/configuration/pull/5488. It's unknown when this issue was introduced though.

See also related PR which added a default for `RETIREMENT_SERVICE_USER_NAME` to edx-platform:    https://github.com/edx/edx-platform/pull/22087 This also appears to be the only other usage of this variable across the edx repositories. Also that PR failed to explain why the change was necessary.

**JIRA tickets**: [https://openedx.atlassian.net/browse/OSPR-3935](OSPR-3935)


**Dependencies**: None

**Sandbox URL**: 

The following sandbox was deployed using this configuration branch:

-    LMS: https://se-1541-test.stage.opencraft.hosting/
 -  Studio: https://studio.se-1541-test.stage.opencraft.hosting/


**Merge deadline**: ASAP - this is breaking provisioning that was previously working

**Testing instructions**:

1. enable the retirement service setup with `COMMON_RETIREMENT_SERVICE_SETUP: true` See also https://github.com/open-craft/configuration/blob/samuel/oauth-retirement-vars-fix/playbooks/roles/user_retirement_pipeline/README.md for retirement specific config possibly required.
2. provision a master devstack
3. verify that it provisions successfully.

**Author notes and concerns**:

- https://github.com/edx/edx-platform/pull/22087 should be at least partially reverted since they introduce vars not used anywhere else.

**Reviewers**

**Settings**
```yaml
```
